### ISSUE TITLE
Discuss adding JS Linting

### ISSUE THREAD

**Issue by gaboesquivel** [EOSIO/eosjs#268]
I noticed there's no eslint config or specific code conventions specified. 
I want to propose adopting standardjs.com as the official code style guideline.

I can help out making the code based compliant and adding lint scripts and precommit hooks

**Comment by DanailMinchev** [comment]
This is great idea, but I would like to recommend **Airbnb JavaScript Style** -  https://github.com/airbnb/javascript as it is more comprehensive in my opinion (and if we search the web).

**Comment by chris-allnutt** [comment]
@DanailMinchev @gaboesquivel we currently use airbnb js and ts linting on all internal projects.  If someone would like to add airbnb to this project as a PR we'd be happy to review and evaluate it.  Updating the travis.yml to run it as part of the build would also be super cool.

I think at this time this code base is fairly expansive so it may be better as an incremental update without being required to pass the build but be part of it.

**Comment by DanailMinchev** [comment]
@chris-allnutt I've created https://github.com/EOSIO/eosjs/pull/275 with initial integration, but it is nor part of the build. If #275 is fine then we can modify the configuration further and fix corresponding files if you think it's a good way to integrate it as an incremental update.

@chris-allnutt @gaboesquivel please let me know if you have other ideas or if I am missing something.

Thanks

**Comment by gaboesquivel** [comment]
@chris-allnutt All public projects seems to be more close to standardjs, the wallet is standardjs actually https://github.com/EOSIO/eos-wallet-app/blob/master/.stylelintrc . 

Adding standardjs code linting to this project is quick. Here's a PR for it
It also adds a pre-commit hook that lints just the git staged files, so it's fast and efficient. https://github.com/EOSIO/eosjs/pull/276

**Comment by gaboesquivel** [comment]
@DanailMinchev lol, we were working on the same thing the same time. With different standards tho. As I mentioned on my previous comment, all public code on github.com/eosio is more close to the StandardJS Format which is also widely used. 

It was fast to make this code base compliant to the StandardJS format.

AirBnb is not bad, It might require more work to make the code compliant to it, I'm not sure. 

Regardless the standard we chose I think the `no-semis` should be maintained, we are all happily coding without them xD.

**Comment by DanailMinchev** [comment]
@gaboesquivel yeah, I just noticed that as well :D

Yes, I mentioned as well in my PR that it is initial integration and to fix the code to be fully compliant I am happy to discuss and apply the changes.

Yes, StandardJS is less work and easier to integrate, but I have my own observation that it has less rules and in my own experience wasn't that useful as other styles (personal opinion), that is why I suggested to try AirBnb.

I don't mind if we go with any of those standards as long as we have one.
